### DISCUSSION TITLE
Implementing Google Auth without NextAuth while utilizing Backend API to get user information using Google API

### DISCUSSION THREAD

**Discussion by Incredible311** [nextauthjs/next-auth#12101] [Help]
```
    "next": "^14.2.14",
    "next-auth": "^5.0.0-beta.20",
```

For now, I customized handler for clicking the button for "Google Sign In" like below.

```
const handleGoogleLogin = (e) => {
  e.preventDefault();

  const googleAuthUrl = `https://accounts.google.com/o/oauth2/v2/auth?` +
    `client_id=${process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID}&` +
    `redirect_uri=${encodeURIComponent('http://localhost:3000/api/oauth/google')}&` +
    `response_type=code&` +
    `access_type=offline&` +
    `scope=openid email profile`;

  popup.current = window.open(googleAuthUrl, 'Google Login', 'width=600,height=600');
};
```

In 'api/oauth/google/route.ts', I defined GET API endpoint to work as redirect_uri for Google Sign In. In this function, I'm calling the backend API endpoint which will get user information using Google API with code from redirect_uri.

I can get accessToken and user information successfully, but have some troubles to update the NextAuth Session so that those information can be used like a normal session in NextAuth.

Please let me know if there is any unclear parts or if you have any concerns.
Looking forward to hearing from you soon.

Best,

**Comment by k-taro56** [comment]
Since you are not using next-auth at all, we cannot update the next-auth session.
